hawkmoth/docstring.py

```diff
@@ -164,6 +149,24 @@ def get_docstring(self, transform=None):
                                   type_spacer=spacer, args=args)
         lines[:0] = header.splitlines()
```

Comment:
I thought this slice was a bit obfuscating before, but now it's pretty bad. Basically we imply here that `_insert_header_lines` is called _after_ we've indented the remaining lines and _before_ we nest them. Both of which occur outside of this method.

Maybe do a `_get_header_lines` instead? That way the final line stitching is done all in the same place.

Reply:
Wasn't too happy with it myself.

Any idea if there's a perf impact for doing this:

```
lines = header.splitlines() + lines
```

instead of:

```
lines[:0] = header.splitlines()
```

?

Reply:
[Didn't test it myself, but seems there isn't any.](https://stackoverflow.com/questions/12088089/python-list-concatenation-efficiency).

This would be my rough take on `get_docstring`:

```python
        header = self._get_header_lines()
        text = self._get_comment_lines()

        # This could be inside `_get_comment_lines`
        self._remove_comment_markers(text)

        if transform is not None:
            ...

        self._nest(text, self._indent)
        lines = header + text
        self._nest(lines, self._nest)

        ...
        return lines
```

Reply:
Any preference on calling the static methods via self or Docstring?

Reply:
Hmm, I wrote `self` mostly out of habit. Only reason I can think of is that if it ever stops being static, `self` will still work and only external users would need fixing.

Reply:
Uh, I hit some issue with `self`, and just left the `Docstring` in for now.